src/status_im/contexts/keycard/backup/events.cljs

```diff
@@ -131,7 +132,9 @@
                {:on-complete init-card-with-pin-and-continue}]])]})))
 
 (rf/reg-event-fx :keycard/backup.phrase-entered
- (fn [{:keys [db]} [{:keys [phrase]}]]
-   {:db (assoc-in db [:keycard :backup :masked-phrase] phrase)
-    :fx [[:dispatch [:navigate-back]]
-         [:dispatch [:open-modal :screen/keycard.backup.ready-to-add]]]}))
+ (fn [{:keys [db]} [{:keys [key-uid seed-phrase]}]]
+   (if (not= key-uid (get-in db [:profile/profile :key-uid]))
+     {:fx [[:dispatch [:enter-seed-phrase/set-error (i18n/label :t/seed-phrase-dont-match)]]]}
```

Comment:
wouldn't it better to pass :profile/profile :key-uid into enter-phrase screen instead?

Reply:
Actually, I tried that first, but the code became very complicated since the error logic changes depending on the flow. So, I think it's better to keep the enter-seed-phrase logic focused solely on entering the seed phrase. The validation logic should be handled within their respective namespaces, as we were already doing for the 'account already exists' scenario.